### DISCUSSION TITLE
Pylance无法识别某些类方法

### DISCUSSION THREAD

**Discussion by zhaosu99** [metab0t/PyOptInterface#8] [中文Q&A] [answered]
您好！ 我对你的包很感兴趣，最近在尝试用。我目前发现了一个小问题。可能poi (pyoptinterface)与 pyoptinterface.highs.Model的有些属性是动态生成的？比如：

model = highs.Model()
model.add_linear_constraint(...) 

这段代码中，pyplance无法识别add_linear_constraint这个类方法。可能这个方法是用C++代码动态生成的？

但这样会提高在撰写代码时的错误率，因为pylance无法识别该方法，无法自动补全，或者将之用代表类方法的颜色高亮显示。

不知道您有没有计划解决这个小问题。

同样的问题也出现在poi.VariableDomain

**Comment by metab0t** [comment]
这个问题因为很多方法是由c++代码实现的，可能需要通过nanobind 2.0的stub代码生成python接口来解决，后续我会研究怎么实现。

另外，有关使用的问题请在Discussion中提出，谢谢！

**Comment by metab0t** [comment, marked as answer]
@zhaosu99

The latest 0.2.4 version has added support for type stubs, and it will help IDE and typecheckers to autocomplete your code.

Welcome to try it and provide feedbacks.
